### DISCUSSION TITLE
data for testing

### DISCUSSION THREAD

**Discussion by valgit** [Rodemfr/MicronetToNMEA#39] [Technical discussions]
Is there any data available for testing ? eg: some data to inject and do a non regression testing ?

**Comment by Rodemfr** [comment]
Hi valgit,

I understand that you would like to test new MicronetToNMEA SW versions against non-regression on your own setup. If I'm correct, that's not an easy question. Today, I'm manually testing the SW by bringing my displays near my PC and by simulating input data with Sailaway and qtVlm/OpenCPN. Not ideal...

Is that what you want to do ?

**Comment by valgit** [comment]
more or less yes. I would like to be able to test the micronet part without having to have a tacktik just near, but same would also be interesting with NMEA & co.
on some professional project, I have mock software to inject data from stream for eg.

**Comment by Rodemfr** [comment]
Unfortunately no. Testing MTN without a display would mean developping a dedicated device listening and checking MTN's behavior on the RF channel. This would double the work compared to MTN. Not something I can afford for a small project like this.
I could add something to help testing but with a display, a new menu sending ramping data to every field of Micronet displays but this would not be very different of the current manual tests I'am passing.

**Comment by valgit** [comment]
In my case, my problem is that I have not always accessed to Micronet devices (shared people device ;-) so I would like to have something like a Micronet dump log to be able to replay it. (somewhat like what wireshark does)

**Comment by Rodemfr** [comment]
I fear this will not be possible with MicronetToNMEA.
